icu4c/source/common/dictbe.cpp

```diff
@@ -1369,7 +1415,12 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
         if (utextPos > prevUTextPos) {
             // Boundaries are added to foundBreaks output in ascending order.
             U_ASSERT(foundBreaks.size() == 0 || foundBreaks.peeki() < utextPos);
-            foundBreaks.push(utextPos, status);
+            if(!foundBreaks.contains(utextPos)
+                    && (utextPos != rangeStart
+                    || (utextPos == rangeStart  && isClosePunct(utext_char32At(inText,utextPos-1))))) {
```

Comment:
remove extra " " before "&&", you should have only one " " between "rangeStart" and "&&", not two.
add " " between "inText," and "utextPos".
add " " before and after "-"